databusclient/client.py

```
@@ -205,6 +205,40 @@ def create_distribution(
 
+def create_distributions_from_metadata(metadata):
+    distributions = []
+    counter = 0
+    for entry in metadata:
+        filename = entry["filename"]
+        checksum = entry["checksum"]
+        size = entry["size"]
+        url = entry["url"]
+        # Expect a SHA-256 hex digest (64 chars). Reject others.
+        if not isinstance(checksum, str) or len(checksum) != 64:
+            raise ValueError(f"Invalid checksum for {filename}: expected SHA-256 hex (64 chars), got '{checksum}'")
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)